src/game/stats.c

```diff
@@ -537,3 +525,38 @@ void Stats_ShowTotal(const char *filename)
     Text_RemoveAll();
     Music_Stop();
 }
+
+void Stats_ComputeTotal(GAMEFLOW_LEVEL_TYPE level_type)
+{
+    memset(&m_TotalStats, 0, sizeof(TOTAL_STATS));
+
+    int16_t secret_flags = 0;
+
+    for (int i = 0; i < g_GameFlow.level_count; i++) {
+        if (g_GameFlow.levels[i].level_type != level_type) {
+            continue;
+        }
+        const GAME_STATS *stats = &g_GameInfo.current[i].stats;
+
+        m_TotalStats.player_kill_count += stats->kill_count;
+        m_TotalStats.player_pickup_count += stats->pickup_count;
+        secret_flags = stats->secret_flags;
+        for (int j = 0; j < MAX_SECRETS; j++) {
+            if (secret_flags & 1) {
+                m_TotalStats.player_secret_count++;
+            }
+            secret_flags >>= 1;
+        }
+
+        m_TotalStats.timer += stats->timer;
+        m_TotalStats.death_count += stats->death_count;
+        m_TotalStats.total_kill_count += stats->max_kill_count;
+        m_TotalStats.total_secret_count += stats->max_secret_count;
+        m_TotalStats.total_pickup_count += stats->max_pickup_count;
+    }
+}
+
+bool Stats_CheckAllSecretsCollected()
```

Comment:
Let's add `(void)` to make it not possible to call the function with arbitrary arguments.